catt/controllers.py

```diff
@@ -184,25 +189,39 @@ def clear(self):
 
 class Cache(CattStore):
     def __init__(self):
-        cache_path = Path(tempfile.gettempdir(), "catt_cache", "chromecast_hosts")
+        cache_path = Path(tempfile.gettempdir(), "catt_%s_cache" % CATT_VERSION, "chromecast_hosts")
         super(Cache, self).__init__(cache_path)
         self._create_store_dir()
 
         if not self.store_path.is_file():
-            devices = pychromecast.get_chromecasts()
-            self._write_store({d.name: d.host for d in devices})
+            devices = get_chromecasts(fail=False)
+            cache_data = {d.name: self._create_device_entry(d.host, d.port) for d in devices}
+            self._write_store(cache_data)
+
+    def _create_device_entry(self, ip, port):
+        device_data = {"ip": ip}
+        if port != DEFAULT_PORT:
+            device_data["group_port"] = port
+        return device_data
 
     def get_data(self, name):
         data = self._read_store()
         # In the case that cache has been initialized with no cc's on the
         # network, we need to ensure auto-discovery.
         if not data:
-            return None
-        # When the user does not specify a device, we need to make an attempt
-        # to consistently return the same IP, thus the alphabetical sorting.
-        if not name:
-            return data[min(data, key=str)]
-        return data.get(name)
+            return (None, None)
+        if name:
+            fetched = data.get(name)
+        else:
+            # When the user does not specify a device, we need to make an attempt
+            # to consistently return the same IP, thus the alphabetical sorting.
+            fetched = data[min(data, key=str)]
+        return (fetched["ip"], fetched.get("group_port")) if fetched else (None, None)
+
+    def set_data(self, name, ip, port):
```

Comment:
Since we're 3-only now, can you add some types here? It helps with mypy. I'll add pre-commit so this is checked on CI as well.

Reply:
Oh we already added `pre-commit`, I'll see about adding mypy to it. The problem with having tens of projects is that you forget what you did where :/

Reply:
Ok

Reply:
I've had to comment out the baseclass ```set_data```  method in order to avoid getting this ```mypy``` error:
```
catt/controllers.py:221: error: Signature of "set_data" incompatible with supertype "CattStore"
```
(adding ```-> None``` does not help)
I realize that this might be bad OOP design anyway, so I'd appreciate any suggestions.

Reply:
I think this is a mypy bug. I added a type ignore on those lines for now.